vault-core-it/vault-core-integration-tests/src/main/java/org/apache/jackrabbit/vault/packaging/integration/ReferenceableIdentifiersImportIT.java

```diff
@@ -401,6 +422,138 @@ public void testImportDupPolicyLegacy() throws RepositoryException, IOException,
         assertNull(duplicateNode);
     }
 
+    @Test
+    public void testInstallPackage_CREATE_NEW_ID() throws Exception {
+        test(IdConflictPolicy.CREATE_NEW_ID, null, null, true, true);
+    }
+
+    @Test
+    public void testInstallPackage_FAIL() throws Exception {
+        test(IdConflictPolicy.FAIL, RepositoryException.class, null, false, false);
+    }
+
+    @Test
+    public void testInstallPackage_FORCE_REMOVE_CONFLICTING_ID() throws Exception {
+        test(IdConflictPolicy.FORCE_REMOVE_CONFLICTING_ID, null, null, false, false);
+    }
+
+    @Test
+    public void testInstallPackage_LEGACY() throws Exception {
+        test(IdConflictPolicy.LEGACY, RepositoryException.class, IllegalStateException.class, false, false);
+    }
+
+    private void test(IdConflictPolicy policy, Class<?> expectedException, Class<?> expectedRootCause, boolean expectNewId,
+            boolean expectRenamedNodeKept) throws Exception {
+
+        String TEST_ROOT = "testroot";
+
+        Node testRoot = getNodeOrNull("/" + TEST_ROOT);
+        if (testRoot == null) {
+            testRoot = admin.getRootNode().addNode(TEST_ROOT);
+        }
+
+        String srcName = String.format("%s-%x.txt", policy, System.nanoTime());
+        String srcPath = PathUtil.append(testRoot.getPath(), srcName);
+
+        Node asset = testRoot.addNode(srcName, NodeType.NT_FOLDER);
+        addFileNode(asset, "binary.txt");
+
+        asset.addMixin(NodeType.MIX_REFERENCEABLE);
+        admin.save();
+
+        String id1 = asset.getIdentifier();
+
+        File pkgFile = exportContentPackage(srcPath);
+
+        String dstPath = srcPath + "-renamed";
+        admin.move(srcPath, dstPath);
+        assertNodeMissing(srcPath);
+
+        try {
+            installContentPackage(pkgFile, policy);
+        } catch (Exception ex) {
+            if (expectedException == null) {
+                throw ex;
+            } else {
+                assertTrue("expected: " + expectedException + ", but got: " + ex.getClass(), expectedException.isInstance(ex));
+                if (expectedRootCause != null) {
+                    Throwable rc = ExceptionUtils.getRootCause(ex);
+                    assertTrue("expected: " + expectedRootCause + ", but got: " + rc.getClass(), expectedRootCause.isInstance(rc));
+                }
+                // expected exception -> test done
+                return;
+            }
+        }
+
+        if (expectRenamedNodeKept) {
+            assertNodeExists(dstPath);
+        } else {
+            assertNodeMissing(dstPath);
+        }
+
+        assertNodeExists(srcPath);
+        assertNodeExists(srcPath + "/binary.txt");
+
+        Node asset2 = testRoot.getNode(srcName);
+        String id2 = asset2.getIdentifier();
+        if (expectNewId) {
+            assertNotEquals(id1, id2);
+        } else {
+            assertEquals(id1, id2);
+        }
+    }
+
+    private Node addFileNode(Node parent, String name) throws Exception {
```

Comment:
why not using https://jackrabbit.apache.org/api/2.20/org/apache/jackrabbit/commons/JcrUtils.html#putFile-javax.jcr.Node-java.lang.String-java.lang.String-java.io.InputStream-?

Reply:
Done.